File: domain/src/main/java/boaz/lol/co/domains/account/dto/AccountCreate.java
package boaz.lol.co.domains.account.dto;

import lombok.Builder;
import lombok.Getter;

@Getter
public class AccountCreate {
    private final String email;
    private final String password;

    @Builder
    public AccountCreate(
            String email,
            String password
    ) {
        this.email = email;
        this.password = password;
    }
}

File: domain/src/main/kotlin/boaz/lol/co/domains/account/AccountCreate.kt
package boaz.lol.co.domains.account

data class AccountCreate (
    val email: String,
    var password: String,
    val riotName: String,
    val riotTag: String,
    val riotId: String,
    val isAgreeToPersonalInfo: Boolean,
    val isAgreeToEmail: Boolean,
    val isCertificatedEmail: Boolean
) {
    fun encryptPassword(encodedPassword: String) {
        this.password = encodedPassword
    }
}